#15163 Repro: Cannot view SQL question when accessing via dashboard with filters connected to modified card without SQL permissions

nemanjaglumac opened the pull request:
### Status
PENDING REVIEW

### What does this PR accomplish?
- Reproduces #15163 in two versions (as per the repro steps)

### How to test this manually?
- `yarn test-cypress-open --spec frontend/test/metabase/scenarios/question/native.cy.spec.js`
- Replace `it.skip()` with `it.only()` to run the test in isolation
- The test should fail until the related issue is fixed

### Additional notes:
- Once the issue is fixed, please remove the `.skip` part (unskip the test completely)
- Make sure the test is passing and
- Merge it together with the fix

### Screenshots:
This test will fail on waiting for `@dataset` request because it never occurs (due to the bug). I tried to assert on the error message on the screen, but test always gives false positives.
![image](https://user-images.githubusercontent.com/31325167/111180730-9aed2000-85ad-11eb-86c8-633b7fff3875.png)

`NOSQL` version of the test
![image](https://user-images.githubusercontent.com/31325167/111210107-1f9b6680-85cd-11eb-9320-5c56691f1804.png)


Sanity check - the same test for admin
![image](https://user-images.githubusercontent.com/31325167/111180996-d4259000-85ad-11eb-9c4a-9e076587ef55.png)

nemanjaglumac committed:
Add repro for #15163

flamber reviewed (commented):
This is good, but there also needs to be tests for step 8+9, since the error is slightly different (and shows the SQL editor, which it shouldn't either), that's the behavior seen when using Sandboxing.

nemanjaglumac committed:
Expand test to include "nosql" scenario

nemanjaglumac commented:
> This is good, but there also needs to be tests for step 8+9, since the error is slightly different (and shows the SQL editor, which it shouldn't either), that's the behavior seen when using Sandboxing.

Updated the test with the second case, and added the assertion the the request's body error. Also, shorten the timeout for this request to 5s, instead of the default 30s.

flamber commented on frontend/test/metabase/scenarios/question/native.cy.spec.js:
```diff
@@ -374,4 +376,90 @@ describe("scenarios > question > native", () => {
+          cy.wait("@dataset", { timeout: 5000 }).then(xhr => {
+            expect(xhr.response.body.error).not.to.exist;
+          });
+          cy.findByText("Showing 1 row");
```
Would this work to make sure editor is not displayed?
```suggestion
          cy.get(".ace_content").should("not.be.visible");
          cy.findByText("Showing 1 row");
```

nemanjaglumac replied:
For sure.
I used slightly different assertion - making sure the element doesn't exist. In certain cases it can really be not visible so let's make sure it's not there at all.

nemanjaglumac committed:
Add additional check for native query editor
